sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/synccommit/SyncCommitMLS.java

```diff
@@ -1,72 +1,31 @@
 package org.sdase.commons.server.kafka.consumer.strategies.synccommit;
 
-import io.prometheus.client.SimpleTimer;
 import java.util.Collections;
 import java.util.Map;
 import org.apache.kafka.clients.consumer.CommitFailedException;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
-import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.clients.consumer.ConsumerRecords;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
 import org.sdase.commons.server.kafka.consumer.ErrorHandler;
-import org.sdase.commons.server.kafka.consumer.KafkaHelper;
 import org.sdase.commons.server.kafka.consumer.MessageHandler;
-import org.sdase.commons.server.kafka.consumer.StopListenerException;
 import org.sdase.commons.server.kafka.consumer.strategies.MessageListenerStrategy;
+import org.sdase.commons.server.kafka.consumer.strategies.autocommit.AutocommitMLS;
 import org.sdase.commons.server.kafka.exception.ConfigurationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /** {@link MessageListenerStrategy} that uses sync commit explicitly before polling a new chunk */
-public class SyncCommitMLS<K, V> extends MessageListenerStrategy<K, V> {
+public class SyncCommitMLS<K, V> extends AutocommitMLS<K, V> {
```

Comment:
Oh, wow. Good catch! A lot of lines saved! 👏

Reply:
One of the rare cases where Sonar could help and the duplicate code warning makes sense. Actually, this is just the quickest solution. I would prefer something with delegation over inheritance but that would cause much more changes. I already was in doubt if this change could bloat the PR too much.